### ISSUE TITLE
[챕터 1] space_stretching 실습에서 torch.no_grad()의 역할

### ISSUE THREAD

**Issue by yskim0** [yskim0/NYU-DEEP-LEARNING#2]
쥬피터 노트북의 Linear transformation with PyTorch 부분 중

```py
with torch.no_grad(): # Why?
    Y = model(X) # Y -> output of my model
    show_scatterplot(Y, colors)
    plot_bases(model(OI))
```

코드에서 `torch.no_grad()` 의 정확한 의미가 궁금합니다.
히스토리 추적을 피하기 위해 사용한다고 하는데, **무슨 히스토리를 담고 있는지/그것을 왜 피해야 하는지** 알려주신다면 감사하겠습니다.

**Comment by leahincom** [comment]
해당 코드가 쓰인 모델

<code>
model = nn.Sequential(
        nn.Linear(2, 2, bias=False)
) </code> <br> <br>

에

> 만약 .requires_grad 속성을 True 로 설정하면, 그 tensor에서 이뤄진 모든 연산들을 추적(track)하기 시작합니다. 계산이 완료된 후 .backward() 를 호출하여 모든 변화도(gradient)를 자동으로 계산할 수 있습니다. 이 Tensor의 변화도는 .grad 속성에 누적됩니다.
기록을 추적하는 것(과 메모리를 사용하는 것)을 방지하기 위해, 코드 블럭을 with torch.no_grad(): 로 감쌀 수 있습니다. 이는 특히 변화도(gradient)는 필요없지만, requires_grad=True 가 설정되어 학습 가능한 매개변수를 갖는 모델을 평가(evaluate)할 때 유용합니다.

를 참고해보면,

requires_grad=True가 default인 점(생략됨)과 bias=False를 이용하여 matrix multiplication을 수행한 점을 통해 이러한 **matrix 연산 과정이 추적 가능하도록 설정됨(히스토리)** 을 알 수 있습니다. 여기서 기록을 추적하는 것을 방지하므로써 해당 코드에서는 gradient를 활용하고 있지 않으므로 **메모리 낭비를 막는 동시에 (학습 가능한 매개변수를 갖는) 모델을 평가하고 있습니다.**

**Comment by yskim0** [comment]
no_grad()로 감싼 코드가 Gradient를 활용하지 않아서 필요없는거군요! 자세한 답변 감사합니다 😄 👍
